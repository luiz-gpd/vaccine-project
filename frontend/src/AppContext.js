import { createContext } from 'react'

const AppContext = createContext();


// const reducer = (state, action) => {
//     switch (action.type) {
//         case "EXEMPLE": {
//             return {
//                 ...state,
//                 loggedUser: action.payload,
//             };
//         }

//         default: {
//             return state;
//         }
//     }
// }

export default AppContext