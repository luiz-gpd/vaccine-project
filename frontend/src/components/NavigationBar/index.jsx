import { useHistory, useLocation } from 'react-router'
import ClayNavigationBar from '@clayui/navigation-bar'
import ClayLink from '@clayui/link'
import React from 'react'
import Agendamento from '../../pages/Agendamento'
import Lista from '../../pages/Lista'

const NavigationBar = () => {

    const routes = [
        {
          component: Agendamento,
          name: 'Agendamento',
          path: '/user',
          private: false
        },
        {
          component: Lista,
          name: 'Lista',
          path: '/list',
          private: false
        },
        
      ];

    const history = useHistory();
    const location = useLocation();

  return (
    <ClayNavigationBar triggerLabel="Site Pitang">
      {routes.map(({ name, path }) => (
        <ClayNavigationBar.Item key={path} className="dynamic-route" active={path === location.pathname}>
          <ClayLink data-testid="link" onClick={() => history.push(path)} className="nav-link" displayType="unstyled">
            {name}
          </ClayLink>
        </ClayNavigationBar.Item>
      ))}
    </ClayNavigationBar>
  );
};

export default NavigationBar;