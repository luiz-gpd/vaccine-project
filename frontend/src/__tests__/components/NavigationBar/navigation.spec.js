import NavigationBar from "../../../components/NavigationBar";

import AppContext, { initialState } from "../../../AppContext";

import { render, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

const NavigationBarWrapper = ({ state = initialState, dispatch }) => (
    <AppContext.Provider value={[state, dispatch]}>
        <BrowserRouter>
            <NavigationBar />
        </BrowserRouter>
    </AppContext.Provider>
);

describe("NavigationBar", () => {
    it("renders", () => {
        const { asFragment } = render(<NavigationBarWrapper />);

        expect(asFragment()).toMatchSnapshot();
    });

    it("renders with context data", () => {
        const { container } = render(<NavigationBarWrapper />);

        expect(container.querySelectorAll(".dynamic-route")).toHaveLength(2);
        
        // const navLink = container.querySelectorAll(".nav-link");

        // fireEvent.click(navLink);
    });
});