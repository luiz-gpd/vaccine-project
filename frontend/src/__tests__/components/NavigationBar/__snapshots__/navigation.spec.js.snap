// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-collapse-absolute navbar-expand-md navbar-underline navigation-bar navigation-bar-light"
  >
    <div
      class="container-fluid container-fluid-max-xl"
    >
      <button
        aria-expanded="false"
        class="navbar-toggler navbar-toggler-link collapsed btn btn-unstyled"
        type="button"
      >
        <span
          class="navbar-text-truncate"
        >
          Site Pitang
        </span>
        <svg
          class="lexicon-icon lexicon-icon-caret-bottom"
          role="presentation"
        >
          <use
            xlink:href="#caret-bottom"
          />
        </svg>
      </button>
      <div
        class="navbar-collapse collapse"
      >
        <div
          class="container-fluid container-fluid-max-xl"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item dynamic-route"
            >
              <a
                class="nav-link link-unstyled"
                data-testid="link"
              >
                Agendamento
              </a>
            </li>
            <li
              class="nav-item dynamic-route"
            >
              <a
                class="nav-link link-unstyled"
                data-testid="link"
              >
                Lista
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
